bin/main

```diff
@@ -1,163 +1,110 @@
-      @game_on_going = false
-      if @current_player == 'O'
-        puts "#{@player2} wins"
-      else
-        puts "#{@player1} wins" 
-      end
+  def display_board
+    puts @board[0] + '|' + @board[1] + '|' + @board[2]
+    puts @board[3] + '|' + @board[4] + '|' + @board[5]
+    puts @board[6] + '|' + @board[7] + '|' + @board[8]
   end
-end
 
-def check_row
-  if @board[0] == @board[1] && @board[0] == @board[2] && @board[0] != '_'
-    @game_on_going = false
-    if @current_player == 'O'
-      puts "#{@player2} wins"
-    else
-      puts "#{@player1} wins" 
+  def players_turn
+    puts 'choose from 1 - 9'
+    turn = gets.chomp.to_i
+    arr = (1..9)
+    until arr.include? turn
+      display_board
+      puts 'invalid choose from 1 - 9'
+      turn = gets.chomp.to_i
     end
-  elsif @board[3] == @board[4] && @board[3] == @board[5] && @board[3] != '_'
-      @game_on_going = false
-      if @current_player == 'O'
-        puts "#{@player2} wins"
-      else
-        puts "#{@player1} wins" 
-      end
-  elsif @board[6] == @board[7] && @board[6] == @board[8] && @board[6] != '_'
-    @game_on_going = false
-    if @current_player == 'O'
-      puts "#{@player2} wins"
-    else
-      puts "#{@player1} wins"
+    turn -= 1
+    while @board[turn] != '_'
+      display_board
+      puts 'Slot already taken or invalid, choose from 1 - 9'
+      turn = gets.chomp.to_i
+      turn -= 1
     end
+    @board[turn] = @current_player
   end
-end
 
-def check_diagonal
-  if @board[0] == @board[4] && @board[0] == @board[8] && @board[0] != '_'
-    @game_on_going = false
-    if @current_player == 'O'
-      puts "#{@player2} wins"
+  def flip_player
+    if @current_player == 'X'
+      @current_player = 'O'
+      puts "It's #{@player2}'s turn to play"
     else
-      puts "#{@player1} wins"
-    end
-  elsif @board[2] == @board[4] && @board[2] == @board[6] && @board[2] != '_'
-    @game_on_going = false
-    if @current_player == 'O'
-      puts "#{@player2} wins"
-    else
-      puts "#{@player1} wins"
+      @current_player = 'X'
+      puts "It's #{@player1}'s turn to play"
     end
   end
-end
 
-def if_tie
-  result = @board.all? do |item|
-    item == 'X' || item == 'O'
+  def check_win
+    validate = GameLogic.new(@board, @current_player)
+    @game_on_going = false if validate.validate_wins(0, 3, 6, @Player2, @player1)
+    @game_on_going = false if validate.validate_wins(1, 4, 7, @Player2, @player1)
+    @game_on_going = false if validate.validate_wins(2, 5, 8, @Player2, @player1)
+    @game_on_going = false if validate.validate_wins(0, 1, 2, @Player2, @player1)
+    @game_on_going = false if validate.validate_wins(3, 4, 5, @Player2, @player1)
+    @game_on_going = false if validate.validate_wins(6, 7, 8, @Player2, @player1)
+    @game_on_going = false if validate.validate_wins(0, 4, 8, @Player2, @player1)
+    @game_on_going = false if validate.validate_wins(2, 4, 6, @Player2, @player1)
```

Comment:
did you try with `||` instead of assigning the variable multiple times??
If you improve your validate method, this could get a lot easier.